pkg/schema/schema.go

```diff
@@ -317,3 +337,80 @@ func validateResources(directory string) error {
 func CheckCatalogResources(manifestDir string) error {
 	return validateResources(manifestDir)
 }
+
+func validateExamplesAnnotations(csv *v1alpha1.ClusterServiceVersion) error {
+	var examples []CRDTemplate
+	annotations := csv.ObjectMeta.GetAnnotations()
+	// Return right away if no alm-examples are found.
+	if annotations == nil {
+		return fmt.Errorf("Unable to get annotations")
+	}
+
+	almExamples, ok := annotations["alm-examples"]
+	if !ok {
+		return fmt.Errorf("Unable to get alm-examples annotations")
+	}
+
+	if json.Unmarshal([]byte(almExamples), &examples) == nil {
+		return fmt.Errorf("Unable to unmarshal alm-examples annotations")
+	}
+
+	err := validateKubectlable([]byte(almExamples))
+	if err != nil {
+		return err
+	}
+
+	providedAPIs, err := getProvidedAPIs(csv)
+	if err != nil {
+		return err
+	}
+	parsedExamples, err := parseExamplesAnnotations(examples)
+	if err != nil {
+		return err
+	}
+
+	if matchGVKProvidedAPIs(parsedExamples, providedAPIs) != nil {
+		return err
+	}
+
+	return nil
+}
+
+func parseExamplesAnnotations(examples []CRDTemplate) (map[opregistry.APIKey]struct{}, error) {
+	parsed := map[opregistry.APIKey]struct{}{}
+	for _, value := range examples {
+		parts := strings.SplitN(value.APIVersion, "/", 2)
+		if len(parts) < 2 {
+			return nil, fmt.Errorf("couldn't parse group/version from crd kind: %s", value.Kind)
+
+		}
+		parsed[opregistry.APIKey{Group: parts[0], Version: parts[1], Kind: value.Kind, Plural: ""}] = struct{}{}
+	}
+	return parsed, nil
+}
+
+func matchGVKProvidedAPIs(examples map[opregistry.APIKey]struct{}, providedAPIs map[opregistry.APIKey]struct{}) error {
+	for key := range examples {
+		if _, ok := providedAPIs[key]; !ok {
+			return fmt.Errorf("couldn't match %v in provided APIs list: %v", key, providedAPIs)
+		}
+	}
+	return nil
+}
+
+func getProvidedAPIs(csv *v1alpha1.ClusterServiceVersion) (map[opregistry.APIKey]struct{}, error) {
+	provided := map[opregistry.APIKey]struct{}{}
+
+	for _, owned := range csv.Spec.CustomResourceDefinitions.Owned {
+		parts := strings.SplitN(owned.Name, ".", 2)
+		if len(parts) < 2 {
+			return nil, fmt.Errorf("couldn't parse plural.group from crd name: %s", owned.Name)
+		}
+		provided[opregistry.APIKey{Group: parts[1], Version: owned.Name, Kind: owned.Kind, Plural: ""}] = struct{}{}
```

Comment:
I'd expect this line to look more like this:
https://github.com/operator-framework/operator-lifecycle-manager/blob/b2d1cd21368bc8cc10e4ca11a231f09077630c33/pkg/controller/registry/resolver/operators.go#L277